clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/agent/KubernetesControllersCachingAgent.groovy

```diff
@@ -223,6 +235,23 @@ class KubernetesControllersCachingAgent extends KubernetesV1CachingAgent impleme
     credentials.apiClientAdaptor.getPods(serverGroup.namespace, serverGroup.selector)
   }
 
+  V1beta1StatefulSet loadStatefulSet(String namespace, String name) {
+    credentials.apiClientAdaptor.getStatefulSet(name, namespace)
+  }
+
+  V1beta1DaemonSet loadDaemonSet(String namespace, String name) {
+    credentials.apiClientAdaptor.getDaemonSet(name, namespace)
+  }
+
+  Boolean isControllerAgentSet(namespace, name) {
+    if (loadStatefulSet(namespace, name)) {
```

Comment:
This still leaves us with the problem of replica sets and daemonset/statefulsets sharing names...

Reply:
That is an obsolete code.  I remove it.